packages/components/src/form/inputs/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,121 @@
+import { appliedStyles, cns } from '@stcland/utils'
+import { path } from 'ramda'
+import { useFormContext } from 'react-hook-form'
+import { RadioButtonGroupStyles } from 'src/styles/componentTypes'
+import { nullFormContext } from '../Form'
+import FormLabel from '../FormLabel'
+
+//*****************************************************************************
+// Interface
+//*****************************************************************************
+type RadioButtonShape = {
+  label: string;
+  value: string;
+};
+
+export interface RadioButtonGroupProps {
+  name: string;
+  label?: string;
+  direction?: 'vertical' | 'horizontal';
+  defaultChecked?: string;
+  radioButtons: RadioButtonShape[];
+  readOnly?: boolean;
+  required?: boolean;
+  disabled?: boolean;
+  hidden?: boolean;
+  customStyles?: Partial<RadioButtonGroupStyles>;
+}
+
+const toId = function (name: string, radioButtonValue: string): string {
+  return name + '.' + radioButtonValue
+}
+
+const RadioButtonGroup = function ({
+  name,
+  label,
+  radioButtons,
+  disabled = false,
+  hidden = false,
+  direction = 'horizontal',
+  defaultChecked = '',
+  required = false,
+  readOnly = false,
+  customStyles,
+}: RadioButtonGroupProps) {
+  if (hidden) return null
+  const { register, formState, clearErrors, setFocus } =
+    useFormContext() || nullFormContext
+  const { errors } = formState || {}
+
+  const errorTextPath = [...name.split('.'), 'message']
+  const errorText: React.ReactNode = path(errorTextPath, errors)
+
+  const registerRequired = {
+    value: required,
+    message: ' is required',
+  }
+  const flexDir = direction === 'horizontal' ? 'flex-row' : 'flex-col'
+
+  const cn: RadioButtonGroupStyles = appliedStyles(
+    {
+      root: 'mb-2 flex gap-2 flex-col',
+      buttonContainer: cns('flex', flexDir, 'gap-2'),
+      label: '',
+      radioContainer: 'flex gap-2',
+      error: 'text-red-300 italic text-sm',
+      radioButton: (row: number) =>
+        cns(
+          'accent-sky-500 h-4 w-4',
+          direction === 'horizontal' && row !== 0 && 'flex-3'
+        ),
+    },
+    customStyles
+  )
```

Comment:
@ianmarquez same concern here as above, there is a case where the customStyles can completely override the `buttonContainer` and we might lose the `flexDir` and `direction` style, this should be added after the styles are `merged`.

`Button` is a good example.